### PR TITLE
fix: support non-ASCII ens

### PR DESCRIPTION
**Pull request by UncleBill**
Test on https://twitter.com/AinuxA

image

### CODE REVIEW

**Review by guanbinrui — APPROVED** [review]

**Comment by guanbinrui** [comment]
Not working with:

+ `🐈🐈‍⬛🐈.eth`
+ `1⃣4⃣6⃣.eth`

**Comment by UncleBill** [comment]
> Not working with:
> 
> * `🐈🐈‍⬛🐈.eth`
> * `1⃣4⃣6⃣.eth`

Fixed in https://github.com/DimensionDev/Maskbook/pull/4268/commits/c02e668e1187bf5c4e777de56c0eb2a09b618fa7

image

File: packages/maskbook/src/social-network-adaptor/twitter.com/utils/user.ts [hunk 1]
```diff
@@ -38,6 +38,18 @@ export const getTwitterId = () => {
 
 export const getBioDescription = () => {
     const node = bioDescriptionSelector().evaluate()
+    if (node?.children) {
```
Comment (Jack-Works) [inline packages/maskbook/src/social-network-adaptor/twitter.com/utils/user.ts:41]: Please try to reuse this function:

https://github.com/DimensionDev/Maskbook/blob/ad14cf2692615fa04e31bf48cf7cb2621ebf2e22/packages/maskbook/src/social-network-adaptor/twitter.com/utils/fetch.ts#L43-L43

Comment (UncleBill) [reply packages/maskbook/src/social-network-adaptor/twitter.com/utils/user.ts:41]: Good suggestion. I'll try later.

File: packages/maskbook/src/social-network-adaptor/twitter.com/utils/fetch.ts [hunk 2]
```diff
@@ -40,15 +40,19 @@ const parseId = (t: string) => {
     return regexMatch(t, /status\/(\d+)/, 1)!
 }
 
-const serializeToText = (node: ChildNode): string => {
+const parseEmojiFragment = (fragment: string) => {
+    return String.fromCodePoint(...fragment.split('-').map((code) => Number.parseInt(code, 16)))
+}
```
Comment (UncleBill) [inline packages/maskbook/src/social-network-adaptor/twitter.com/utils/fetch.ts:45]: to support https://abs-0.twimg.com/emoji/v2/svg/1f408-200d-2b1b.svg

**Review by guanbinrui — APPROVED** [review]

**Review by nuanyang233 — APPROVED** [review]

**Pull request merged**